arangod/Pregel/ArangoExternalDispatcher.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include "Actor/ActorPID.h"
+#include "Actor/Message.h"
+#include "CrashHandler/CrashHandler.h"
+#include "Assertions/ProdAssert.h"
+#include "Inspection/VPackWithErrorT.h"
+#include "Logger/LogMacros.h"
+#include "Network/ConnectionPool.h"
+#include "Network/Methods.h"
+
+namespace arangodb::pregel {
+
+struct NetworkMessage {
+  actor::ActorPID sender;
+  actor::ActorPID receiver;
+  VPackBuilder payload;
+};
+template<typename Inspector>
+auto inspect(Inspector& f, NetworkMessage& x) {
+  return f.object(x).fields(f.field("sender", x.sender),
+                            f.field("receiver", x.receiver),
+                            f.field("payload", x.payload));
+}
+
+struct ArangoExternalDispatcher {
+  static auto errorHandling(arangodb::network::Response const& message)
+      -> arangodb::ResultT<VPackSlice> {
+    if (message.fail()) {
+      return {arangodb::Result{
+          TRI_ERROR_INTERNAL,
+          fmt::format("REST request failed: {}",
+                      arangodb::fuerte::to_string(message.error))}};
+    }
+    if (message.statusCode() >= 400) {
+      return {arangodb::Result{
+          TRI_ERROR_FAILED,
+          fmt::format("REST request returned an error code {}: {}",
+                      message.statusCode(), message.slice().toJson())}};
+    }
+    return message.slice();
+  }
+
+  ArangoExternalDispatcher(std::string url,
+                           network::ConnectionPool* connectionPool)
+      : connectionPool(connectionPool), baseURL{std::move(url)} {}
+
+  auto send(actor::ActorPID sender, actor::ActorPID receiver,
+            arangodb::velocypack::SharedSlice msg) -> network::FutureRes {
+    auto networkMessage =
+        NetworkMessage{.sender = sender,
+                       .receiver = receiver,
+                       .payload = velocypack::Builder(msg.slice())};
+    auto serialized = inspection::serializeWithErrorT(networkMessage);
+    ADB_PROD_ASSERT(serialized.ok());
+    auto builder = velocypack::Builder(serialized.get().slice());
+    return network::sendRequestRetry(
+        this->connectionPool,
+        // TODO: what about "shard:"?
```

Comment:
what about `shard:`?

Reply:
Yes, I'm not yet sure how to solve that. `shard:` is used (instead of `server:`) in OutgoingCache, here I think the receiving server is not known. In the actor framework we can currently address actors only via server id, so here we possibly need to change something.